ui-tests/tst/software/aws/toolkits/jetbrains/uitests/fixtures/AwsExplorer.kt

```diff
@@ -5,44 +5,94 @@ package software.aws.toolkits.jetbrains.uitests.fixtures
 
 import com.intellij.remoterobot.RemoteRobot
 import com.intellij.remoterobot.data.RemoteComponent
+import com.intellij.remoterobot.fixtures.CommonContainerFixture
+import com.intellij.remoterobot.fixtures.ComponentFixture
 import com.intellij.remoterobot.fixtures.FixtureName
 import com.intellij.remoterobot.search.locators.byXpath
 import com.intellij.remoterobot.stepsProcessing.step
+import com.intellij.remoterobot.utils.waitFor
+import org.assertj.swing.timing.Pause
 import java.time.Duration
+import kotlin.random.Random
 
-fun RemoteRobot.awsExplorer(
+fun IdeaFrame.awsExplorer(
     timeout: Duration = Duration.ofSeconds(20),
     function: AwsExplorer.() -> Unit
 ) {
+    val locator = byXpath("//div[@accessiblename='AWS Explorer Tool Window' and @class='InternalDecorator']")
+
     step("AWS explorer") {
-        find<AwsExplorer>(byXpath("//div[@class='ExplorerToolWindow']"), timeout).apply(function)
+        val explorer = try {
+            find<AwsExplorer>(locator, timeout)
+        } catch (e: Exception) {
+            step("Open tool window") {
+                // Click the tool window stripe
+                find(ComponentFixture::class.java, byXpath("//div[@accessiblename='AWS Explorer' and @class='StripeButton' and @text='AWS Explorer']")).click()
+                find<AwsExplorer>(locator, timeout)
+            }
+        }
+
+        explorer.apply(function)
     }
 }
 
 @FixtureName("AWSExplorer")
 open class AwsExplorer(
     remoteRobot: RemoteRobot,
     remoteComponent: RemoteComponent
-) : DialogFixture(remoteRobot, remoteComponent) {
+) : CommonContainerFixture(remoteRobot, remoteComponent) {
+    private fun explorerTree() = find<JTreeFixture>(byXpath("//div[@class='Tree']"))
+
     fun openExplorerActionMenu(vararg path: String) {
-        findExplorerTree().rightClickPath(*path)
+        explorerTree().rightClickPath(*path)
     }
 
     fun expandExplorerNode(vararg path: String) {
-        findExplorerTree().expandPath(*path)
-        // wait for loading to disappear
-        try {
-            while (true) {
-                findText("loading...")
-                Thread.sleep(100)
+        expandServiceNode(path.first())
+
+        explorerTree().expandPath(*path)
+        explorerTree().waitUntilLoaded()
+    }
+
+    private fun expandServiceNode(serviceName: String) {
+        step("Expand service node '$serviceName'") {
+            for (attempt in 1..MAX_ATTEMPTS) {
+                val explorerTree = explorerTree()
+
+                waitFor { explorerTree.hasPath(serviceName) }
+                explorerTree.expandPath(serviceName)
+                explorerTree.waitUntilLoaded()
+
+                step("DEBUG: ${explorerTree.findAllText().joinToString { it.text }}") {}
```

Comment:
is there a debug logger?

Reply:
I'll delete it when I am done with this PR.